### PR TITLE
add onBlur handler to Formik component

### PR DESCRIPTION
**Pull request by FranciscoMSM**

### CODE REVIEW

File: src/Formik.tsx [hunk 1]
```diff
@@ -483,7 +484,11 @@ export class Formik<Values = FormikValues> extends React.Component<
       }));
 
       if (this.props.validateOnBlur) {
-        this.runValidations(this.state.values);
+        this.runValidations(this.state.values).then(combinedErrors => {
+          if (this.props.onBlur && field) {
```
Comment (davidgomes) [inline src/Formik.tsx:488]: Do you know when there wouldn't be a `field` here?

**Review by davidgomes — APPROVED** [review]

File: src/Formik.tsx (hunk 1 above)
Comment (FranciscoMSM) [reply src/Formik.tsx:488]: Not in my head, will test a bit

Comment (FranciscoMSM) [reply src/Formik.tsx:488]: Actually this is quite simple, you can have focused elements inside the form other than inputs (divs with tabIndex=0 for instance) which make the field not always present

Comment (davidgomes) [reply src/Formik.tsx:488]: :+1:

**Comment by lneves12** [comment]
How hard would be to backport these changes to the upstream of formik and open a PR? 

IMO if we have a fork with this kind of changes it would be nice to open a PR directly in the main repo as well, otherwise, it can be really hard to upgrade or maintain.

Formik is already on version 2. It shouldn't be too hard for us to upgrade to formik v2 after applying this patch. The only major breaking changes on V2 seems to be depending on react 16.8

**Pull request merged**